src/parser/ast/ASTContext.h

```diff
@@ -174,7 +174,90 @@ typedef Vector<ASTBlockContext *, GCUtil::gc_malloc_atomic_allocator<ASTBlockCon
 typedef std::unordered_map<AtomicString, StorePositiveNumberAsOddNumber, std::hash<AtomicString>, std::equal_to<AtomicString>,
                            GCUtil::gc_malloc_allocator<std::pair<AtomicString const, StorePositiveNumberAsOddNumber>>>
     FunctionContextVarMap;
+struct ASTScopeContextClassPrivateUsingInfo {
+    AtomicString m_name;
+    size_t m_index;
+    size_t m_lineNumber;
+    size_t m_lineStart;
+
+    ASTScopeContextClassPrivateUsingInfo(AtomicString name,
+                                         size_t index,
+                                         size_t lineNumber,
+                                         size_t lineStart)
+        : m_name(name)
+        , m_index(index)
+        , m_lineNumber(lineNumber)
+        , m_lineStart(lineStart)
+    {
+    }
+};
+
+typedef Vector<ASTScopeContextClassPrivateUsingInfo, GCUtil::gc_malloc_atomic_allocator<ASTScopeContextClassPrivateUsingInfo>, ComputeReservedCapacityFunctionWithLog2<>> ASTScopeContextClassPrivateUsingInfoVector;
+
+struct ASTClassInfo {
+    ASTClassInfo *m_parent;
+    ASTClassInfo *m_firstChild;
+    ASTClassInfo *m_nextSibling;
+    AtomicStringVector m_classPrivateNames;
+    ASTScopeContextClassPrivateUsingInfoVector m_classPrivateNameUsingInfo;
+
+    void appendChild(ASTClassInfo *child)
+    {
+        ASSERT(child->m_nextSibling == nullptr);
+        if (m_firstChild == nullptr) {
+            m_firstChild = child;
+        } else {
+            ASTClassInfo *tail = firstChild();
+            while (tail->m_nextSibling != nullptr) {
+                tail = tail->m_nextSibling;
+            }
```

Comment:
If last child(tail) address is stored too, this searching process can be removed.
IMO since `ASTClassInfo` is temporally allocated in the AST memory pool, holding additional tail address seems not a severe overhead. What do you think about it?

Reply:
That's good idea!

Reply:
I updated my pr